rust/src/lib/policy/capture.rs

```diff
@@ -86,6 +88,60 @@ impl NetworkCaptureRules {
     pub(crate) fn is_empty(&self) -> bool {
         self.cmds.is_empty()
     }
+
+    fn sort_captures(&mut self) -> Result<(), NmstateError> {
+        for _ in 0..SORT_CAPTURE_MAX_ROUND {
+            if self.set_capture_priority() {
+                self.cmds
+                    .sort_unstable_by_key(|(_, cmd)| cmd.capture_priority);
+                return Ok(());
+            }
+        }
+        Err(NmstateError::new(
+            ErrorKind::InvalidArgument,
+            format!(
+                "Failed to sort the policy capture in \
+                {SORT_CAPTURE_MAX_ROUND} round of rotation, \
+                please order the capture in desire \
+                policy by placing capture before its consumer"
```

Comment:
User re-roder of the yaml dictionary makes no effect depending on the yaml parser implementation, the only thing that the user can check is the number of references or if there is a loop at them.

Reply:
New unit test included proving user sorted capture matters.